### PR TITLE
gRPC - specify client compression via GrpcClientConfiguration

### LINKED ISSUE #17066: gRPC - consider exposing AbstractStub.withCompression(String) via GrpcClientConfiguration
Right now, a user needs to modify each injected stub manually, e.g.

```java
@Singleton
public class MyBean {

    private final MutinyHelloGrpc.MutinyHelloStub client;

    public MyBean(@GrpcClient("hello") MutinyHelloGrpc.MutinyHelloStub stub) {
       this.client = stub.withCompression("gzip");
    }
```
It might make sense to allow users to specify the default value in the `application.properties` file instead:
```
quarkus.grpc.clients.hello.compression=gzip
```

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #17066

### CODE REVIEW

File: extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcDotNames.java [hunk 1]
```diff
@@ -29,4 +31,7 @@
     static final MethodDescriptor RETRIEVE_CHANNEL_METHOD = MethodDescriptor.ofMethod(Channels.class, "retrieveChannel",
             Channel.class, String.class);
 
+    static final MethodDescriptor ADJUST_STUB = MethodDescriptor.ofMethod(GrpcClientConfigProvider.class,
+            "adjustStub", AbstractStub.class, String.class, AbstractStub.class);
```
Comment (cescoffier) [inline extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcDotNames.java:35]: not sure about "adjust", what about "configure" ?

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/GrpcClientConfiguration.java [hunk 2]
```diff
@@ -143,4 +143,10 @@
      */
     @ConfigItem(defaultValue = "pick_first")
     public String loadBalancingPolicy;
+
+    /**
+     * The compression to use for each call.
```
Comment (cescoffier) [inline extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/GrpcClientConfiguration.java:148]: Can you list the supported values?
Also, I believe we only support gzip in native at the moment (I should be able to get snappy to work too as I did it for Kafka).

**Review by cescoffier — CHANGES_REQUESTED** [review]
Looks good to me, I did two minor comments.

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/GrpcClientConfiguration.java (hunk 2 above)
Comment (mkouba) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/GrpcClientConfiguration.java:148]: It seems that the compressor must be registered in the current `CompressorRegistry` and by default only `gzip` is registered.  And I have no idea if we support a custom `CompressorRegistry` in quarkus.

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
Thanks!

**Pull request merged**